src/main/java/catan/services/util/play/PlayUtil.java

```diff
@@ -0,0 +1,135 @@
+
+import catan.domain.exception.GameException;
+import catan.domain.model.game.GameBean;
+import catan.domain.model.game.GameUserBean;
+import catan.domain.model.game.types.GameStage;
+import catan.domain.model.game.types.GameUserActions;
+import catan.domain.transfer.output.game.AllAvailableActionsDetails;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static catan.services.impl.GameServiceImpl.ERROR_CODE_ERROR;
+
+@Component
+public class PlayUtil {
+    private Logger log = LoggerFactory.getLogger(PlayUtil.class);
+
+    private EndTurnUtil endTurnUtil;
+
+    private static final Gson GSON = new Gson();
+
+    public List<List<String>> getInitialBuildingsSetFromJson(String json) {
+        return GSON.fromJson(json, new TypeToken<List<List<String>>>() {
+        }.getType());
+    }
+
+    public AllAvailableActionsDetails getAllAvailableActions(String json) {
+        return GSON.fromJson(json, AllAvailableActionsDetails.class);
+    }
+
+    public void updateNextMoveOrder(GameBean game) throws GameException {
+        Integer nextMoveNumber;
+        switch (game.getStage()) {
+            case PREPARATION:
+                List<List<String>> initialBuildingsSet = getInitialBuildingsSetFromJson(game.getInitialBuildingsSet());
+                nextMoveNumber = endTurnUtil.endTurnImplInPreparationStage(game, initialBuildingsSet);
+                break;
+            case MAIN:
+                nextMoveNumber = endTurnUtil.endTurnImplInMainStage(game);
+                break;
+            default:
+                log.debug("Cannot recognize current game stage: {}", game.getStage());
+                throw new GameException(ERROR_CODE_ERROR);
+        }
+
+        log.debug("Next move order in {} stage is changing from {} to {}", game.getStage(), game.getCurrentMove(), nextMoveNumber);
+        game.setCurrentMove(nextMoveNumber);
+    }
+
+    public void updateCurrentCycleBuildingNumber(GameBean game) {
+        if (game.getStage().equals(GameStage.PREPARATION)) {
+            List<List<String>> initialBuildingsSet = getInitialBuildingsSetFromJson(game.getInitialBuildingsSet());
+            if (game.getCurrentCycleBuildingNumber() > initialBuildingsSet.get(game.getPreparationCycle() - 1).size()) {
+                game.setCurrentCycleBuildingNumber(1);
+            } else {
+                game.setCurrentCycleBuildingNumber(game.getCurrentCycleBuildingNumber() + 1);
+            }
+        }
+    }
+    public void updateAvailableUserActions(GameBean game) throws GameException {
+        boolean isMandatory = false;
+        List<String> actionsList = new ArrayList<String>();
+        List<GameUserActions> actionCode = new ArrayList<GameUserActions>();
```

Comment:
rename to some normar name qoz it is unreal to understand what happens with variable that has name 'actionCode', but contains LIST of GameUserActions